### ISSUE TITLE
Internal function to check for standardized columns for plots

### ISSUE THREAD

**Issue by kgaonkar6** [d3b-center/annoFuse#14]
# Description
Internal check for files that are provided as input in shiny_fuse or plots and other functions in general just 1 to check all columns. 

# Specifics 
For domain plots basic columns are  "LeftBreakpoint", "RightBreakpoint", "FusionName", "Gene1A" and "Gene1B" 
For recurrent plots and summary plot the basic columns are "LeftBreakpoint", "RightBreakpoint", "FusionName", "Gene1A" and "Gene1B" and a "grouping variable"
For filtering_fusion_QC the basic columns are "LeftBreakpoint" , "RightBreakpoint" , "FusionName" , "Sample" , "Caller" , "Fusion_Type" , "JunctionReadCount"  , "SpanningFragCount" , "annots"

- [x]  @kgaonkar6 Make "Sample" column as user provided ID in annofuse_single_sample and formt_calls functions 
- [x] @kgaonkar6 to check if all columns as read as characters or numeric in input_files

How will work on this
@federicomarini  has added an internal function to check and add message if it correct input file.

**Comment by jharenza** [comment]
To tag onto this, I noticed you are using `Sample` in the results file, so when we go to merge with clinical, we will have to change it back to `Kids_First_Biospecimen_ID` or alternatively rename `Kids_First_Biospecimen_ID` to `Sample` in the clinical file. Is `Sample` a required column in the input? If so, probably leave as such. I suppose alternatively, you can create a variable asking the user to identify which column is the "ID" column, and you can pull the name of the column for use throughout the analysis and then it would be retained for the final output. Just an idea to simplify later steps.

**Comment by kgaonkar6** [comment]
hmm yeah good idea, right now in the standardization step I have "Sample" as hardcoded I can actually look into making it more flexible.

**Comment by federicomarini** [comment]
Makes sense. 
I did start working on such a function to check e.g. all columns are there in the annoFuse-utils.R in `exemplary`.
Feel free to expand it there - keeping in mind all the functions basically do check that everytime 😉

**Comment by kgaonkar6** [comment]
Updating the request to make Sample columnname flexible with this idea from @federicomarini 

if "Sample" is already a column in the input file , check OK. 
If not, user needs to specify which column to be added as "Sample" and use that for all the funcs throughout the package/app

**Comment by kgaonkar6** [comment]
@federicomarini  to check if the input fusion calls (STARFusion and arriba) columns are correctly formatted would it be good to have a generic check like this or should it be specific caller specific checks .. Suggestions? 

Mainly because for smaller arriba files, breakpoints get formatted as "date"! if only run as read_tsv() without specifying to read that as character col_type. 


```
# check column class for input files
.check_input_calls <- function(inputFusioncalls) {

  # all calls have columns as character or numeric
  cols_fusioncalls <- sapply(inputFusioncalls,class)
  stopifnot(all(cols_fusioncalls %in% c("character","numeric")))

  invisible(inputFusioncalls)
}

```

**Comment by federicomarini** [comment]
Oh I see.
What if you read with non-`readr` functions? Say, the good ol' `read.delim`? I like a lot `readr`, but sometimes it makes wild guesses in terms of column types...

But yes: nothing speaks against adding extra checks to make our calls more robust.
If we want to take this "to the next level"... we could even check the input to the standardizer functions, according to the software of provenance.

**Comment by kgaonkar6** [comment]
read.delim has different issue with arriba :P 
it changes the special characters in arriba to "."

#gene1                       gene2        strand1(gene/fusion)        strand2(gene/fusion) 
X.gene1                       gene2        strand1.gene.fusion.        strand2.gene.fusion. 

let me see if I can add per fusion caller checks and hopefully they don't change the format ever again :D

**Comment by federicomarini** [comment]
> it changes the special characters in arriba to "."

ah, I see - it is because they deprecate spaces in the column names.
Can that work if you specify `check.names=FALSE` in the call?

> hopefully they don't change the format ever again :D

my 2 cents will be that it will happen one day, but as long as it is properly documented, we can accommodate that as well

**Comment by jharenza** [comment]
I was just going to comment - second to `check.names=FALSE`

**Comment by kgaonkar6** [comment]
I think read.delim(check.names=FALSE) will work as well if users want to use that specifically. 

Looking at other packages (chimeraviz) they have specific col_types as a function call for the specific caller so I've implemented that in the recent PR. The issue with that I believe is that if the caller's change the column names then that needs to be updated specifically for the annoFuse_single_sample function call

**Comment by federicomarini** [comment]
Probably it is worth mentioning in the documentation of each function the version number it is supposed to work with.
Or version numbers, or ranges of (e.g. >= version x.y.z).

**Comment by kgaonkar6** [comment]
Good idea! I've added the specific version's we've used 
STAR-Fusion 1.5.0
Arrib v1.1.0

I can check if STAR-Fusion format has been the same for some versions for the range

**Comment by kgaonkar6** [comment]
it seems STAR-Fusion has the same format from >=1.50 ( their repo only has wiki pages from v1.6.0 supporting the format)

**Comment by jharenza** [comment]
closed with #56
